### DISCUSSION TITLE
Tagged MOM6 releases

### DISCUSSION THREAD

**Discussion by navidcy** [mom-ocean/MOM6#1587] [General]
Hi all,

I was wondering how come you don't do tagged MOM6 releases. This way when we run a simulation we would know exactly what snapshot of the code we compiled an executable with and this would help when we try to reproduce results?

**Comment by navidcy** [comment]
I'm wondering if https://github.com/NOAA-GFDL/MOM6/discussions is a better place to put this.

I'm a bit confused with the duplicate versions of MOM6 under NOAA-GFDL and mom-ocean. Seems like the NOAA-GFDL is the latest/actively being developed but then the mom-ocean is the one that has all the discussions?

**Comment by marshallward** [comment]
I think mom-ocean is the right place to ask this question.

To answer the second question: 

Yes, NOAA-GFDL/MOM6 is very active, but its `dev/gfdl` branch should be considered a development branch suited for use within GFDL.  The `main` branch at mom-ocean is the version that has been tested and verified by the other partners, and is the closest we have to a public release.

We even have a picture describing this relationship: https://www.marshallward.org/mom6workshop/contrib.html#/the-mom6-consortium

At one time, `dev/gfdl` and `main` were both hosted at NOAA-GFDL/MOM6, but we decided to separate these branches into repositorites, in order to stress the distinction.

As for the first question of versioning: This predates me and I think @adcroft or @Hallberg-NOAA are in a better position to answer it.  But I could see some benefit in tagging each merge from their respective nodes to `main`.

**Reply by navidcy** [reply]
Thanks @marshallward!

Regarding tagging: it doesn't have to be _every_ merge. Only when bug fixes are in place or breaking features are introduced, e.g., in accordance with [semantic versioning](https://semver.org).

**Reply by navidcy** [reply]
If *this* (the `mom-ocean`) is the official version that is tested and verified, then we need to fix the installation instructions?

At the moment the installation instructions point users to `MOM6-examples` and there the instructions guide us to clone the `NOAA-GFDL` repo. That's why I have always, up to this point, thought that the `NOAA-GFDL/MOM6` was the main one. :)

**Reply by marshallward** [reply]
I don't know if we need to go full semver.  Our definitions of stability revolve around floating point reproducibility rather than API compatibility, and we guarantee backwards compatibility for a very long time (if not perpetually) so it may not be a good fit.  But it deserves discussion, and we might be able to follow something similar.  We could raise this at a future MOM6 dev call (biweekly).

**Reply by marshallward** [reply]
BTW we are currently handling #1586 which is as close as we have to a release.  These happen roughly ever 1-2 months, so that's the frequency we'd be talking about.

**Reply by navidcy** [reply]
1-2 months is great!

**Reply by marshallward** [reply]
Re: MOM6-examples, this is nominally a collection of GFDL tests, so probably ought to be used with the NOAA-GFDL/MOM6 codebase.  It also happens to be the reference set of GFDL regression tests, so it may not even work with mom-ocean codebase (e.g. new parameters not yet added to mom-ocean).

Which is another way of saying that you're right :laughing: We do need find a more general introduction to the model.  We've discussed some ideas, but have found it difficult to reconcile them with the testing and other internal uses.  Until then, we're stuck with many arrows pointing to NOAA-GFDL/MOM6.
